curated_transformers/pipeline/distill.py

```diff
@@ -168,5 +169,31 @@ def initialize(self, get_examples, *, nlp=None):
         assert len(doc_sample) > 0, Errors.E923.format(name=self.name)
         self.model.initialize(X=doc_sample)
 
-    def pipe(self, stream: Iterable[Doc], *, batch_size: int = 128) -> Iterator[Doc]:
-        return iter(stream)
+    def predict(self, docs: Iterable[Doc]) -> Any:
+        return None
+
+    def set_annotations(self, docs: Iterable[Doc], scores) -> None:
+        pass
+
+    def update(
+        self,
+        examples: Iterable["Example"],
+        *,
+        drop: float = 0.0,
+        sgd: Optional[Optimizer] = None,
+        losses: Optional[Dict[str, float]] = None,
+    ) -> Dict[str, float]:
+        if losses is None:
+            losses = {}
+        losses.setdefault(self.name, 0.0)
+        return losses
+
+    def finish_update(self, sgd: Optimizer) -> None:
+        pass
+
+    def add_label(self, label: str) -> int:
```

Comment:
We should probably not override this one, so that we get an error when someone tries to override this.

Reply:
Just `add_label`?

Reply:
Good question! Maybe `update` too, because a user (like `Language`) should just not call this method when `is_trainable == False`.